niworkflows/workflows/epi/refmap.py

```diff
@@ -0,0 +1,220 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+"""Workflow for the generation of EPI (echo-planar imaging) references."""
+
+from nipype.pipeline import engine as pe
+from nipype.interfaces import utility as niu
+
+from ...engine.workflows import LiterateWorkflow as Workflow
+
+
+DEFAULT_MEMORY_MIN_GB = 0.01
+
+
+def init_epi_reference_wf(
+    omp_nthreads,
+    auto_bold_nss=False,
+    name="epi_reference_wf",
+):
+    """
+    Build a workflow that generates a reference map from a set of EPI images.
+
+    .. danger ::
+
+        All input files MUST have the same shimming configuration.
+        At the very least, make sure all input EPI images are acquired within the
+        same session, and have the same PE direction and total readout time.
+
+    Inputs to this workflow might be a list of :abbr:`SBRefs (single-band references)`,
+    a list of fieldmapping :abbr:`EPIs (echo-planar images)`, a list of
+    :abbr:`BOLD (blood-oxygen level-dependent)` images, or a list of
+    :abbr:`DWI (diffusion-weighted imaging)` datasets.
+    Please note that these different modalities should be mixed together in any
```

Comment:
Unless I'm reading this wrong...

```suggestion
    Please note that these different modalities should not be mixed together in any
```